react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+export default class HotButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { clicks: 0 };
+    this.clickCount = this.clickCount.bind(this);
+  }
+
+  clickCount() {
+    this.setState(prevState => ({
+      clicks: prevState.clicks + 1
+    }));
+  }
+
+  render() {
+    if (this.state.clicks <= 3) {
+      return <button className="purple" onClick={this.clickCount}>Hot Button</button>;
+    } else if (this.state.clicks <= 6) {
+      return <button className="light-purple" onClick={this.clickCount}>Hot Button</button>;
+    } else if (this.state.clicks <= 9) {
+      return <button className="red" onClick={this.clickCount}>Hot Button</button>;
+    } else if (this.state.clicks <= 12) {
+      return <button className="orange" onClick={this.clickCount}>Hot Button</button>;
+    } else if (this.state.clicks <= 15) {
+      return <button className="yellow" onClick={this.clickCount}>Hot Button</button>;
+    } else {
+      return <button className="white" onClick={this.clickCount}>Hot Button</button>;
+    }
```

Comment:
This creates a lot of repetition and chances for bugs and inconsistencies. It would be best to use the `if` statements to determine the correct class, then have a single `return` at the bottom. For example:
```JSX
let btnClass = '';

if (this.state.clicks <= 3) {
  btnClass = 'purple';
} else if (this.state.clicks <= 6) {
  btnClass = 'light-purple';
} else if (this.state.clicks <= 9) {
  btnClass = 'red';
} else if (this.state.clicks <= 12) {
  btnClass = 'orange';
} else if (this.state.clicks <= 15) {
  btnClass = 'yellow';
} else {
  btnClass = 'white';
}

return <button className={btnClass} onClick={this.clickCount}>Hot Button</button>;
```

Reply:
![Screen Shot 2022-09-29 at 5 51 06 PM](https://user-images.githubusercontent.com/44100814/193166782-efd48a1b-a911-4de8-bfe1-802aedb0ac99.png)